### PR TITLE
api: Document `version` in `/build`

### PR DESCRIPTION
**Pull request by vvoland**
It was introduced in API v1.38 but wasn't documented.

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

File: api/swagger.yaml [hunk 1]
```diff
@@ -8331,6 +8331,15 @@ paths:
...
           default: ""
+        - name: "version"
+          in: "query"
+          type: "string"
+          default: "1"
+          description: |
+            Version of the underlying builder to use.
+
+            "1" is the first generation classic builder in docker daemon (default).
+            "2" is a moby/buildkit based builder.
```
Comment (dvdksn) [inline api/swagger.yaml:8342]: just some nit suggestions

```suggestion
            Version of the builder backend to use.

            - `1` is the first generation classic builder in the Docker daemon (default)
            - `2` is [BuildKit](https://github.com/moby/buildkit)
```

Comment (vvoland) [reply api/swagger.yaml]: Thanks, changed!

**Review by dvdksn — APPROVED** [review]
LGTM

**Review by sam-thibault — APPROVED** [review]

File: api/swagger.yaml [hunk 2]
```diff
@@ -8331,6 +8331,15 @@ paths:
...
           type: "string"
           default: ""
+        - name: "version"
+          in: "query"
+          type: "string"
+          default: "1"
+          description: |
+            Version of the builder backend to use.
+
+            - `1` is the first generation classic builder in the Docker daemon (default)
```
Comment (rumpl) [inline api/swagger.yaml:8341]:
```suggestion
            - `1` is the first generation classic (deprecated) builder in the Docker daemon (default)
```

Comment (vvoland) [reply api/swagger.yaml:8341]: Do we want to call it deprecated if it's default?

Comment (rumpl) [reply api/swagger.yaml:8341]: If you try and build with the classic builder you get

```
$ DOCKER_BUILDKIT=0 docker build ....
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            BuildKit is currently disabled; enable it by removing the DOCKER_BUILDKIT=0
            environment-variable.
```

Comment (rumpl) [reply api/swagger.yaml:8341]: But yeah, the default being the deprecated one isn't really ideal I'll give you that

Comment (thaJeztah) [reply api/swagger.yaml:8341]: I think we're not printing that message for native Windows (as there's no buildkit support yet), so it's a bit of a "in between state".

I'm personally fine with leaving the `(deprecated)` for a follow-up, but also ok with doing it now 🤷 

We could also add a separate block below the list of options which would give a bit more room to outline;

```
Deprecated: .....
```

File: api/swagger.yaml [hunk 3]
```diff
@@ -8331,6 +8331,15 @@ paths:
...
+            - `2` is [BuildKit](https://github.com/moby/buildkit)
```
Comment (thaJeztah) [inline api/swagger.yaml:8342]: Does Swagger / OpenAPI v2 allow enums for query parameters? Looks like it does; perhaps we should use that


```yaml
        - name: "foo"
          in: "query"
          description: "BuildKit output configuration"
          type: "string"
          enum:
            - "hello"
            - "world"
          default: "hello"
```


Screenshot 2024-02-01 at 15 56 31

File: api/swagger.yaml (hunk 2 above)
Comment (vvoland) [reply api/swagger.yaml]: Added `(deprecated`).

File: api/swagger.yaml (hunk 3 above)
Comment (vvoland) [reply api/swagger.yaml:8343]: Oh thanks, done!

**Review by rumpl — APPROVED** [review]

**Review by thaJeztah — APPROVED** [review]
LGTM

**Pull request merged**